app/models/participant.rb

```diff
@@ -141,6 +141,10 @@ def first_middle
     [first_name, middle_initial].join(' ')
   end
 
+  def can_be_destroyed?
+    appointments.map{|appointment| appointment.procedures}.flatten.detect{|procedure| !procedure.unstarted?}.nil?
```

Comment:
`procedures.where.not(status: 'unstarted').any?`

This does a single count query